monai/metrics/compute_meandice.py

```diff
@@ -74,6 +74,6 @@ def compute_meandice(y_pred,
     y_pred_o = torch.sum(y_pred, reduce_axis)
     denominator = y_o + y_pred_o
```

Comment:
Adding a very small smoothing value to the denominator would take care of the divide-by-zero issue, see the Dice loss definition: https://github.com/Project-MONAI/MONAI/blob/822d5686cb6e56452954c165d470ede7212d96a6/monai/losses/dice.py#L79

Reply:
> Adding a very small smoothing value to the denominator would take care of the divide-by-zero issue, see the Dice loss definition:
> 
> https://github.com/Project-MONAI/MONAI/blob/822d5686cb6e56452954c165d470ede7212d96a6/monai/losses/dice.py#L79

Hi @ericspod ,

We should **not compute** MeanDice if labels of this channel are all zero. Dice metric is only useful for positive predictions.
I think adding a smooth will cause the value error and not expected.
Thanks.

Reply:
If you calculate the mean dice for zero labels you would expect to get zero as the metric value, at least I'd expect that. Adding a tiny smoothing factor, say 1e-10, is way below the margin of error we'd expect anyway so it'll have no impact on accuracy but prevent divide-by-zero.

Reply:
Hi @ericspod ,

If the **preds and labels are all zero**, your example code:
```py
 score = 2.0 * (intersection.sum(1) + smooth) / (sums.sum(1) + smooth) 
```
will be 2.0 x (0.0 + 1e-10) / (0.0 + 1e-10) = 2.0
I think it's not expected value, right?
Anyway, from NV research's previous work, they always skip all zero labels for MeanDice.
Thanks.

Reply:
I think we could make the smoothing factor a user specified argument in both metric and loss

Reply:
If we don't add it to intersection it'll work for metrics. This was the training loss where there wouldn't ever be blank segmentations.

Reply:
Hi @wyli and @ericspod ,

I think you guys can make a decision here whether to merge this PR.
Many NV side researchers told me to skip the zero labels for MeanDice,
it's different from DiceLoss. And that's what we did in Clara.
Thanks.